datafusion/core/src/datasource/physical_plan/mod.rs

```diff
@@ -482,6 +508,171 @@ fn get_projected_output_ordering(
     all_orderings
 }
 
+// A normalized representation of file min/max statistics that allows for efficient sorting & comparison.
```

Comment:
I think it would help to explain the requirements on this structure for correctness:
1. That all rows in the sort order have statistics for all values